q2_feature_classifier/_blast.py

```diff
@@ -0,0 +1,88 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2017, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import tempfile
+import pandas as pd
+from os.path import join
+from q2_types.feature_data import (
+    FeatureData, Taxonomy, Sequence, DNAFASTAFormat)
+from qiime2.plugin import Int, Str, Float, Choices, Range
+from .plugin_setup import plugin
+from ._consensus_assignment import (_consensus_assignments,
+                                    _get_default_unassignable_label,
+                                    _run_command)
+
+
+def blast(query: DNAFASTAFormat, reference_reads: DNAFASTAFormat,
+          reference_taxonomy: pd.Series, maxaccepts: int=10, min_id: float=0.8,
+          strand: str='both', evalue: float=0.001,
+          min_consensus: float=0.51,
+          unassignable_label: str=_get_default_unassignable_label(),
+          num_threads: str=1) -> pd.DataFrame:
+
+    min_id = min_id * 100
+    seqs_fp = str(query)
+    ref_fp = str(reference_reads)
+    with tempfile.TemporaryDirectory() as blastdir:
+        blastref = join(blastdir, 'blastref')
+        # make temporary blast db
+        cmd = ['makeblastdb', '-in', ref_fp, '-out', blastref, '-dbtype',
```

Comment:
Evidently you are the 💥 📄 , because that works @BenKaehler . Thanks! Now I know who to ask when I'm too lazy to peruse the blast documentation more carefully in the future!